src/ApiPort/CommandLine/AnalyzeOptions.cs

```diff
@@ -191,10 +192,23 @@ public int Compare(FileInfo x, FileInfo y)
                     }
 
 
-                    return x.FullName.CompareTo(y?.FullName);
+                    return x.Path.CompareTo(y?.Path);
```

Comment:
It seems like we should rely on the IAssemblyFile implementation to define the comparer. Comparing paths works for FilePathAssemblyFile, but I can imagine IAssemblyFile implementations that represent assemblies in memory or something like that. In those cases, paths might always be null and there would have to be some other mechanism for comparison (perhaps that IAssemblyFile implementation would define an assembly name field and use that, for example).